### PR TITLE
Fixes the issue that nativeTest cannot be executed using Junit 5.11.0-M2

### LINKED ISSUE #602: Unable to execute nativeTest under Junit 5.11.0-M2
**Before reporting**

- This repository should be used to report **issues on the Maven or Gradle plugins for GraalVM**.
- Please report issues which are specific to [the Spring Framework](https://spring.io/) or [the Micronaut framework](https://micronaut.io/) to their specific repositories.
- Do not report issues with building your specific application, e.g errors which happen at image build time like classes initialized at build time, or missing classes as run time: those are not related to the plugins but problems with configuration. You can refer to the [GraalVM native image documentation](https://www.graalvm.org/latest/reference-manual/native-image/) for available options and the [plugins documentation](https://graalvm.github.io/native-build-tools) for how to use them with the plugin.

**Describe the bug**
**Make sure that you have read [the documentation](https://graalvm.github.io/native-build-tools) and that you are using the latest plugin version.**

- Unable to execute nativeTest under Junit 5.11.0-M2.
- I created a git with minimal unit tests at https://github.com/linghengqian/junit-v5110-test .

**To Reproduce**

When possible, provide a link to a repository which reproduces the issue, with instructions on how to use.
The reproducer **must** make use of either the Maven or Gradle plugin.

- Execute the following command on the Ubuntu 22.04.3 instance with `SDKMAN!` installed.
```bash
sdk install java 22.0.1-graalce
sdk use java 22.0.1-graalce
sudo apt-get install build-essential libz-dev zlib1g-dev -y

git clone git@github.com:linghengqian/junit-v5110-test.git
cd ./junit-v5110-test/
./mvnw -PnativeTestInJunit -T1C -e clean test
```
Please use backticks to [properly format code](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting).
If possible please attach a complete reproducer here (either as [a zip file](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files) or as a link to public repository/branch).

**Expected behavior**

**Logs**
Add logs to help explain your problem. 
Please use backticks to [properly format big logs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#fenced-code-blocks). 

```bash
[2/8] Performing analysis...  [****]                                                                    (11.8s @ 0.69GB)
    6,818 reachable types   (81.3% of    8,388 total)
    9,294 reachable fields  (51.0% of   18,209 total)
   30,315 reachable methods (52.4% of   57,859 total)
    2,358 types,    72 fields, and   883 methods registered for reflection
       58 types,    58 fields, and    52 methods registered for JNI access
        4 native libraries: dl, pthread, rt, z

Error: An object of type 'org.junit.platform.commons.util.LruCache' was found in the image heap. This type, however, is marked for initialization at image run time for the following reason: classes are initialized at run time by default.
This is not allowed for correctness reasons: All objects that are stored in the image heap must be initialized at build time.

You now have two options to resolve this:

1) If it is intended that objects of type 'org.junit.platform.commons.util.LruCache' are persisted in the image heap, add 

    '--initialize-at-build-time=org.junit.platform.commons.util.LruCache'

to the native-image arguments. Note that initializing new types can store additional objects to the heap. It is advised to check the static fields of 'org.junit.platform.commons.util.LruCache' to see if they are safe for build-time initialization,  and that they do not contain any sensitive data that should not become part of the image.

2) If these objects should not be stored in the image heap, you can use 

    '--trace-object-instantiation=org.junit.platform.commons.util.LruCache'

to find classes that instantiate these objects. Once you found such a class, you can mark it explicitly for run time initialization with 

    '--initialize-at-run-time=<culprit>'

to prevent the instantiation of the object.

If you are seeing this message after upgrading to a new GraalVM release, this means that some objects ended up in the image heap without their type being marked with --initialize-at-build-time.
To fix this, include '--initialize-at-build-time=org.junit.platform.commons.util.LruCache' in your configuration. If the classes do not originate from your code, it is advised to update all library or framework dependencies to the latest version before addressing this error.

The following detailed trace displays from which field in the code the object was reached.
Detailed message:
Trace: Object was reached by
  reading field java.util.Collections$SynchronizedMap.m of constant 
    java.util.Collections$SynchronizedMap@2633a885: {}
  reading static field org.junit.platform.commons.util.ReflectionUtils.interfaceMethodCache
    at <unknown-location>
  registered as read because: null
```

**System Info (please complete the following information):**
 - OS: [e.g. `Windows`] `Ubuntu 22.04.4`
 - GraalVM Version [e.g. `22.0 CE`] `24.0.1 CE`
 - Java Version [e.g. `17`] `22`
 - Plugin version [e.g. `native-gradle-plugin:0.9.10`] `native-maven-plugin:0.10.2`

**Additional context**

- More early discussion is at https://github.com/apache/shardingsphere/issues/29052 .

### PR DESCRIPTION
**Pull request by linghengqian**
- Fixes #602 .
- Fixes the issue that nativeTest cannot be executed using Junit 5.11.0-M2.
- The project's gradle toolchain actively looks for a random jdk17 release, such as Microsoft OpenJDK 17. Setting up a GraalVM version of JDK11 can be very confusing.

### CODE REVIEW

**Review by linghengqian — COMMENTED** [review]
- @dnestoro @fniephaus Is it possible to temporarily disable the execution of `./gradlew :native-gradle-plugin:functionalTest` on GraalVM CE Dev For JDK24 / https://github.com/graalvm/graalvm-ce-dev-builds/releases/tag/24.1.0-dev-20240612_2110 ? I cannot reproduce the CI results of Github Actions locally, refer to https://github.com/graalvm/native-build-tools/actions/runs/9517556127/job/26236415815?pr=603 .
```
sdk install java 17.0.11-graal
sdk use java 17.0.11-graal

git clone git@github.com:linghengqian/native-build-tools.git -b fix-junit
cd ./native-build-tools/
./gradlew :native-gradle-plugin:functionalTest
```

File: .sdkmanrc [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # Enable auto-env through the sdkman_auto_env config
 # Add key=value pairs of SDKs to use below
-java=22.2.r11-grl
+java=17.0.11-graal
```
Comment (vjovanov) [inline .sdkmanrc:3]: Why did we need this change?

File: .github/workflows/test-native-gradle-plugin.yml [hunk 2]
```diff
@@ -89,7 +89,7 @@ jobs:
       - name: "🔧 Install GraalVM (dev)"
         uses: graalvm/setup-graalvm@main
         with:
-          java-version: 'dev'
+          java-version: '22.0.1'
```
Comment (vjovanov) [inline .github/workflows/test-native-gradle-plugin.yml:92]: Also, why this change?

File: .sdkmanrc (hunk 1 above)
Comment (linghengqian) [reply .sdkmanrc:3]: - I initialized a new Ubuntu WSL distribution instance to test the current project, and it turns out that the project requires both GraalVM CE For JDK11 and GraalVM CE For JDK22 to be installed via SDKMAN! to run unit tests. I will roll back this change.

File: .github/workflows/test-native-gradle-plugin.yml (hunk 2 above)
Comment (linghengqian) [reply .github/workflows/test-native-gradle-plugin.yml]: - This is related to the issue I pointed out at https://github.com/graalvm/native-build-tools/pull/603#pullrequestreview-2118567359 . Under GraalVM CE For JDK24 Dev, this CI fails as https://github.com/graalvm/native-build-tools/actions/runs/9517556127/job/26236415815?pr=603 . This CI works fine under GraalVM CE For JDK22, as shown in https://github.com/graalvm/native-build-tools/actions/runs/9520173469/job/26244879976?pr=603 . I don't know anything about the changes after JDK23. 
- I have now rolled back the changes to show the CI.

**Review by vjovanov — APPROVED** [review]
LGTM, thanks!

**Review by dnestoro — APPROVED** [review]

**Comment by dnestoro** [comment]
@linghengqian let me try to fix failing gate

**Review by melix — APPROVED** [review]
